app/components/smart-banner-test.js

```diff
@@ -0,0 +1,190 @@
+		daysHiddenAfterClose: 15,
+
+		// Duration to hide the banner after it is clicked (0 = always show banner)
+		daysHiddenAfterView: 30,
+	},
+	day: 86400000,
+
+	appId: computed.oneWay(`config.appId.android`),
+	appScheme: computed.oneWay(`config.appScheme.android`),
+	config: computed('wikiVariables', function () {
+		return this.get('wikiVariables').get('smartBanner') || {};
+	}),
+	dbName: computed.reads('wikiVariables.dbName'),
+	description: computed.oneWay('config.description'),
+	icon: computed.oneWay('config.icon'),
+	iconSize: 92,
+
+	iconStyle: computed('icon', function () {
+		if (this.get('noIcon')) {
+			return null;
+		}
+
+		const icon = Thumbnailer.getThumbURL(this.get('icon'), {
+			mode: Thumbnailer.mode.thumbnailDown,
+			width: this.iconSize,
+			height: this.iconSize
+		});
+
+		return new htmlSafe(`background-image: url(${icon})`);
+	}),
+
+	link: 'https://itunes.apple.com/us/developer/wikia-inc/id422467077',
+
+	noIcon: computed.not('icon'),
+	title: computed.oneWay('config.name'),
+
+	actions: {
+		/**
+		 * @returns {void}
+		 */
+		close() {
+			this.setSmartBannerCookie(this.get('options.daysHiddenAfterClose'));
+			this.sendAction('toggleVisibility', false);
+			this.track(trackActions.close);
+		},
+
+		/**
+		 * @returns {void}
+		 */
+		view() {
+			const appScheme = this.get('appScheme');
+
+			this.setSmartBannerCookie(this.get('options.daysHiddenAfterView'));
+
+			if (appScheme) {
+				this.tryToOpenApp(appScheme);
+			} else {
+				window.open(this.get('link'), '_blank');
+			}
+
+			this.sendAction('toggleVisibility', false);
+		},
+	},
+
+	/**
+	 * @param {MouseEvent} event
+	 * @returns {void}
+	 */
+	click(event) {
+		const $target = this.$(event.target);
+
+		if (!$target.is('.sb-test-close')) {
+			this.send('view');
+		}
+	},
+
+	/**
+	 * @returns {void}
+	 */
+	willInsertElement() {
+		// this HAVE TO be run while rendering, but it cannot be run on didInsert/willInsert
+		// running this just after render is working too
+		run.scheduleOnce('afterRender', this, this.checkForHiding);
+	},
+
+	/**
+	 * @returns {void}
+	 */
+	checkForHiding() {
+		const {name, disabled} = this.get('config');
+
+		// Show custom smart banner only when a device is Android
+		// website isn't loaded in app and user did not dismiss it already
+		if ((system === 'android' ||
+			// Smart Banner AB Testing
+			this.get('shouldShowABTestBannerOnIOS')) &&
+			// Smart Banner AB Testing
+			!standalone &&
+			name &&
+			!disabled &&
+			$.cookie('sb-closed') !== '1'
+		) {
+			this.sendAction('toggleVisibility', true);
+			this.track(trackActions.impression);
+		}
+	},
+
+	/**
+	 * Try to open app using custom scheme and if it fails go to fallback function
+	 *
+	 * @param {string} appScheme
+	 * @returns {void}
+	 */
+	tryToOpenApp(appScheme) {
+		this.track(trackActions.open);
+		window.document.location.href = `${appScheme}://`;
+
+		run.later(this, this.fallbackToStore, 300);
```

Comment:
could you explain me how does this `run.later` work?

Reply:
=== setTimeout

Reply:
Invokes the passed target/method and optional arguments after a specified period of time. The last parameter of this method must always be a number of milliseconds.

Reply:
@xkxd see https://emberjs.com/api/ember/2.15/namespaces/Ember.run/methods/later?anchor=later and http://www.programwitherik.com/why-you-should-care-about-the-ember-run-loop/